src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,116 +1,105 @@
-import React, { useState } from 'react';
-import { postComment } from '../../api/comments';
-
 import './NewCommentForm.scss';
-
-import { Loader } from '../Loader';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../react-app-env';
 
 type Props = {
-  postId: number,
-  fetchComments: () => void,
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
 };
 
-export const NewCommentForm: React.FC<Props> = React.memo(({
-  postId,
-  fetchComments,
+export const NewCommentForm: React.FC<Props> = ({
+  commentsList, setCommentsList,
 }) => {
-  const [newCommentName, setNewCommentName] = useState('');
-  const [newCommentEmail, setNewCommentEmail] = useState('');
-  const [newCommentBody, setNewCommentBody] = useState('');
-  const [isInputFill, setIsInputFill] = useState(false);
-  const [isLoading, setLoading] = useState(false);
-
-  const removeFillError = () => {
-    if (isInputFill) {
-      setIsInputFill(false);
-    }
-  };
-
-  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setNewCommentName(event.target.value);
-    removeFillError();
-  };
+  const [yourname, setYourname] = useState('');
```

Comment:
The line "useState('')" can be improved by using "useState<string>('')" instead. This is because it explicitly declares the type of the state variable as a string, which can prevent potential errors and improve code readability. By using this syntax, TypeScript will also be able to provide more accurate type checking and better code suggestions.
This applies to the next 3 lines as well